### PR TITLE
feat(api): support variadic arguments on `Table.group_by()`

### PR DESCRIPTION
**Pull request by NickCrews**
Fixes https://github.com/ibis-project/ibis/issues/8504

### CODE REVIEW

File: ibis/expr/types/relations.py [hunk 1]
```diff
@@ -978,9 +978,7 @@ def group_by(
         """
         from ibis.expr.types.groupby import GroupedTable
 
-        if by is None:
-            by = ()
-
+        by = tuple(v for v in by if v is not None)
```
Comment (cpcloud) [inline ibis/expr/types/relations.py:981]: Why wouldn't None be a valid grouping key?

Comment (NickCrews) [reply ibis/expr/types/relations.py:981]: There is an existing test in this same file:

```python
@pytest.mark.parametrize("group", [[], (), None])
def test_group_by_nothing(table, group):
    with pytest.raises(com.IbisInputError):
        table.group_by(group)
```

which is testing for the fairly help error message:

```
    def __init__(self, groupings, **kwargs):
        if not groupings:
>           raise com.IbisInputError("No group keys provided")
E           ibis.common.exceptions.IbisInputError: No group keys provided

ibis/expr/types/groupby.py:48: IbisInputError
```

without this little filter, this test fails with this diffferent, cryptic error:

```
<generator object bind at 0x13ad77de0>) has failed due to the following errors:
E             `groupings`: <generator object bind at 0x13ad77de0> is not a tuple of coercibles to a Value[+T, Columnar]
E           
E           Expected signature: GroupedTable(table: Relation, groupings: tuple[Value[+T, Columnar], ...], orderings: tuple[SortKey, ...] = (), havings: tuple[Value[Boolean, DataShape], ...] = ())
```

If I convert the result of `bind()` into a tuple, it becomes more clear the cause:

```
(None,)) has failed due to the following errors:
E             `groupings`: (None,) is not a tuple of coercibles to a Value[+T, Columnar]
E           
E           Expected signature: GroupedTable(table: Relation, groupings: tuple[Value[+T, Columnar], ...], orderings: tuple[SortKey, ...] = (), havings: tuple[Value[Boolean, DataShape], ...] = ())
```

So maybe we should change the implementation, but I think I want to keep the current public error behavior

Comment (NickCrews) [reply ibis/expr/types/relations.py:981]: this also maybe brings up a side issue that these other more internal errors could be tweaked to be more helpful, but that's separate

Comment (kszucs) [reply ibis/expr/types/relations.py:981]: I think `None` should be valid as well, and `bind()` will turn it to a null literal automatically.

Comment (cpcloud) [reply ibis/expr/types/relations.py:981]: In that case, someone can do a follow up to make the breaking change to `group_by(None)` behavior.

**Review by cpcloud — APPROVED** [review]
Sweet!

**Pull request merged**